docs/tutorials/8_train_custom_env.ipynb

```diff
@@ -0,0 +1,494 @@
+    "\n",
+    "        return self.viewer.render(return_rgb_array=mode == \"rgb_array\")\n",
+    "\n",
+    "    def close(self):\n",
+    "        if self.viewer:\n",
+    "            self.viewer.close()\n",
+    "            self.viewer = None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: create the environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From here, we have two options:\n",
+    "- Add the environment to the gym registry, and use it with existing utilities (e.g. `make`)\n",
+    "- Use the environment directly\n",
+    "\n",
+    "You only need to execute the cells in step 2a, or step 2b to proceed.\n",
+    "\n",
+    "At the end of these steps, we want to have:\n",
+    "- `env`: a single environment that we can use for training an expert with SB3\n",
+    "- `venv`: a vectorized environment where each individual environment is wrapped in `RolloutInfoWrapper`, that we can use for collecting rollouts with `imitation`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2a (recommended): add the environment to the gym registry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The standard approach is adding the environment to the gym registry."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gym.register(\n",
+    "    id=\"custom/FixedHorizonCartPole-v0\",\n",
+    "    entry_point=FixedHorizonCartPoleEnv,  # This can also be the path to the class, e.g. `fixed_horizon_cartpole:FixedHorizonCartPoleEnv`\n",
+    "    max_episode_steps=500,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After registering, you can create an environment is `gym.make(env_id)` which automatically handles the `TimeLimit` wrapper.\n",
+    "\n",
+    "To create a vectorized env, you can use the `make_vec_env` helper function (Option A), or create it directly (Options B1 and B2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from gym.wrappers import TimeLimit\n",
+    "from imitation.data import rollout\n",
+    "from imitation.data.wrappers import RolloutInfoWrapper\n",
+    "from imitation.util.util import make_vec_env\n",
+    "from stable_baselines3.common.vec_env import DummyVecEnv, SubprocVecEnv\n",
+    "\n",
+    "# Create a single environment for training an expert with SB3\n",
+    "env = gym.make(\"custom/FixedHorizonCartPole-v0\")\n",
+    "\n",
+    "\n",
+    "# Create a vectorized environment for training with `imitation`\n",
+    "\n",
+    "# Option A: use the `make_vec_env` helper function - make sure to pass `post_wrappers=[lambda env, _: RolloutInfoWrapper(env)]`\n",
```

Comment:
This is a good illustration! I like the "There Is More Than One Way To Do It" style (while still suggesting the approach that we think is best for common use cases).